agdb_api/typescript/src/query_builder.ts

```diff
@@ -337,19 +372,55 @@ class InsertBuilder {
                         throw new Error("invalid db_id");
                     }
                 } else {
-                    data.values.Multi.push([
-                        {
-                            key: { String: key },
-                            value: elem[key],
-                        },
-                    ]);
+                    let keyValue = convertToDbValue(key);
+                    let valueValue = convertToDbValue(elem[key]);
+                    if (keyValue !== undefined && valueValue !== undefined) {
+                        multiItem.push(
+                            {
+                                'key': keyValue,
+                                'value': valueValue,
+                            }
+                        )
+                    };
                 }
             }
         }
 
         return new InsertValuesIdsBuilder(data);
     }
 
+    elementsAliases(elems: any[], alias: string): InsertNodesValuesBuilder {
```

Comment:
I believe this function is not necessary and you can enhance the existing `InsertBuilder().elements()` where you can apply the same value conversion as you did in the aliases function. The special handling of the `db_id` should be preserved as that is what is used to identify the elements and with it you would not need to use the alias.

In fact looking at that logic in the TS version of the API it should not reject any non-number value like it does now. In the original/Rust version this value is `Option<DbId>` where `DbId` is just a wrapper around `u64`. When the value is `None` (undefined / null) new element is inserted. When the value is `Some` it updates the element being pointed to.

I would therefore suggest to apply the same logic, something like:

```
if (key === "db_id") {
    let id = elem[key];
    if (typeof id === "number") {
        data.ids.Ids.push({ Id: id });
    } else if (id === null || id === undefined) {
        /* intentionally does nothing */
    } else {    
        throw new Error("invalid db_id");
    }
} else {
  let keyValue = convertToDbValue(key);
  let valueValue = convertToDbValue(elem[key]);
  if (keyValue !== undefined && valueValue !== undefined) {
      multiItem.push(
          {
              'key': keyValue,
              'value': valueValue,
          }
      )
  };
}
```

Reply:
The reason for the separation is that in the Schema, both InsertNodesQuery and InsertValuesQuery cannot provide both Ids and aliases at the same time

Reply:
> The reason for the separation is that in the Schema definition, neither InsertNodesQuery nor InsertValuesQuery can simultaneously provide both Ids and aliases.

Yes that is the limitation of the current implementation which we have discussed and will be addressed in #1092 and #1097. I will update the TS API as well with those changes.